### PR TITLE
Add the option to deploy with ssh instead of adb.

### PR DESCRIPTION
**Pull request by wouter182**
Add the option to deploy with ssh instead of adb.
Also some small updates

Pushed here as a draft, because it is not finished but gives already the direction for adding the ssh option

### CODE REVIEW

File: crossbuilder [hunk 1]
```diff
@@ -710,24 +763,26 @@ deploy_deb () {
...
         else
             exec_device /tmp/repo/$CREATE_REPO_SCRIPT /tmp/repo
         fi;
 
-        exec_device "SUDO_ASKPASS=/tmp/askpass.sh sudo -A sed -i '/Pin-Priority/c\Pin-Priority: 50' /etc/apt/preferences.d/extra-ppas.pref"
+        exec_device "SUDO_ASKPASS=/tmp/askpass.sh sudo -A sed -i '/Pin-Priority/c\Pin-Priority: 50' /etc/apt/preferences.d/localrepo.pref"
         exec_device SUDO_ASKPASS=/tmp/askpass.sh sudo -A  apt-get update -o Dir::Etc::sourcelist="/tmp/repo/sources.list" -o Dir::Etc::sourceparts="-" -o APT::Get::List-Cleanup="0"
         exec_device SUDO_ASKPASS=/tmp/askpass.sh sudo -A apt-get dist-upgrade -o Dir::Etc::sourcelist="/tmp/repo/all.list" --yes --force-yes
-        exec_device "SUDO_ASKPASS=/tmp/askpass.sh sudo -A sed -i '/Pin-Priority/c\Pin-Priority: 1001' /etc/apt/preferences.d/extra-ppas.pref"
+        exec_device "SUDO_ASKPASS=/tmp/askpass.sh sudo -A sed -i '/Pin-Priority/c\Pin-Priority: 3000' /etc/apt/preferences.d/localrepo.pref"
```
Comment (wouter182) [inline crossbuilder:784]: I am still in doubt what the pin priority must be

File: crossbuilder [hunk 2]
```diff
@@ -710,24 +763,26 @@ deploy_deb () {
...
-            SERIES=$(adb shell lsb_release -cs | tr -d '\r')
+            SERIES=$(exec_device lsb_release -cs | tr -d '\r')
             exec_device "printf 'Package: *\nPin: release o=local\nPin-Priority: 2000\n\nPackage: *\nPin: release a=$SERIES*\nPin-Priority: 50' | SUDO_ASKPASS=/tmp/askpass.sh sudo -A tee /etc/apt/preferences.d/localrepo.pref"
 #            exec_device SUDO_ASKPASS=/tmp/askpass.sh sudo -A apt-get update
         else
             exec_device /tmp/repo/$CREATE_REPO_SCRIPT /tmp/repo
         fi;
 
-        exec_device "SUDO_ASKPASS=/tmp/askpass.sh sudo -A sed -i '/Pin-Priority/c\Pin-Priority: 50' /etc/apt/preferences.d/extra-ppas.pref"
+        exec_device "SUDO_ASKPASS=/tmp/askpass.sh sudo -A sed -i '/Pin-Priority/c\Pin-Priority: 50' /etc/apt/preferences.d/localrepo.pref"
```
Comment (wouter182) [inline crossbuilder:781]: Changed this to localrepo because extra-ppas didn't exist.

File: crossbuilder [hunk 3]
```diff
@@ -783,13 +854,17 @@ deploy_to_device () {
...
-        adb push $SOURCE_PATH_LOCAL/$POST_DEPLOY_SCRIPT /tmp
+        copy_to_device $SOURCE_PATH_LOCAL/$POST_DEPLOY_SCRIPT /tmp
         exec_device sh /tmp/$(basename $POST_DEPLOY_SCRIPT)
     else
         echo "${POSITIVE_COLOR}If a script named $POST_DEPLOY_SCRIPT existed, it would be executed on device after every deploy.${NC}"
     fi
 }
 
+undeploy_on_device () {
+    echo "Undeployment on device is not working yet"
```
Comment (wouter182) [inline crossbuilder:865]: No idea how to do this at the moment....

**Comment by wouter182** [comment]
This will solve https://github.com/ubports/crossbuilder/issues/42 and https://github.com/ubports/crossbuilder/issues/14

I don't have an android device with ubuntu touch, hence I can not test it.

**Comment by wouter182** [comment]
--no-deb option fails at the moment.

**Comment by mardy** [comment]
I'm closing this since we already merged https://github.com/ubports/crossbuilder/pull/62, but please feel free to reopen if you believe there are more changes that need to go in.

**Pull request closed without merging**